java/memory/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -237,14 +239,25 @@ private ArrowBuf bufferWithoutReservation(final int size, BufferManager bufferMa
 
     final AllocationManager manager = new AllocationManager(this, size);
     final BufferLedger ledger = manager.associate(this); // +1 ref cnt (required)
-    final ArrowBuf buffer = ledger.newArrowBuf(0, size, bufferManager);
+    
+    int offset = calcOffset(ledger.getUnderlying());
+    final ArrowBuf buffer = ledger.newArrowBuf(offset, size, bufferManager);
```

Comment:
When we allocate memory we don't know exact memory location. For sure we need to align size to 64 bytes too. Netty normalize capacity too. But this would not be aligned to 64 bytes. What if we add small overhead to the size.

int alignedSize = size + 63;
alignedSize  += alignedSize % CACHE_ALIGN;